### PR TITLE
feat: Add ordering keys support to Kafka connector

### PR DESCRIPTION
**Pull request by kamalaboulhosn**

### CODE REVIEW

File: kafka-connector/README.md [hunk 1]
```diff
@@ -86,9 +86,9 @@ Connector supports the following configs:
...
+| kafka.partition.scheme | round_robin, hash_key, hash_value, kafka_partitioner, ordering_key | round_robin | The scheme for assigning a message to a partition in Kafka. The scheme "round_robin" assigns partitions in a round robin fashion, while the schemes "hash_key" and "hash_value" find the partition by hashing the message key and message value respectively. "kafka_partitioner" scheme delegates partitioning logic to kafka producer, which by default detects number of partitions automatically and performs either murmur hash based partition mapping or round robin depending on whether message key is provided or not. "ordering_key" uses the hash code of a message's ordering key. If not ordering key is present, uses "round_robin".|
```
Comment (samarthsingal) [inline kafka-connector/README.md:91]: s/If not ordering key is present/If no ordering key is present/g

File: kafka-connector/README.md [hunk 2]
```diff
@@ -108,6 +108,7 @@ Connector supports the following configs:
 | gcp.credentials.json | String | Optional | GCP credentials JSON blob | If specified, use the explicitly handed credentials. Consider using the externalized secrets feature in Kafka Connect for passing the value. |
 | metadata.publish | Boolean | false | When true, include the Kafka topic, partition, offset, and timestamp as message attributes when a message is published to Cloud Pub/Sub. |
 | headers.publish | Boolean | false | When true, include any headers as attributes when a message is published to Cloud Pub/Sub. |
+| orderingKeySource | String | none, key, partition | none | When set to "key", uses a message's key as the ordering key. If set to "partition", converts the partition number to a String and uses that as the ordering key. |
```
Comment (samarthsingal) [inline kafka-connector/README.md:111]: Maybe we should also specify 'none' behavior.

File: kafka-connector/src/main/java/com/google/pubsub/kafka/source/CloudPubSubSourceTask.java [hunk 3]
```diff
@@ -201,8 +210,13 @@ public void start(Map<String, String> props) {
...
+  private SourceRecord createRecordWithHeaders(
+      Map<String, String> messageAttributes,
+      Map<String, String> ack,
+      String key,
+      String orderingKey,
+      byte[] messageBytes,
+      Long timestamp) {
     ConnectHeaders headers = new ConnectHeaders();
     for (Entry<String, String> attribute :
             messageAttributes.entrySet()) {
```
Comment (samarthsingal) [inline kafka-connector/src/main/java/com/google/pubsub/kafka/source/CloudPubSubSourceTask.java:222]: Maybe we should also add the ordering key as an attribute to the message sent to Kafka? Independent of whether we use it to partition or not.

**Review by samarthsingal — APPROVED** [review]
LGTM

File: kafka-connector/src/main/java/com/google/pubsub/kafka/source/CloudPubSubSourceTask.java (hunk 3 above)
Comment (kamalaboulhosn) [reply kafka-connector/src/main/java/com/google/pubsub/kafka/source/CloudPubSubSourceTask.java:227]: Yes, good idea. I have added it in a single commit added to this PR if you want to take a look: https://github.com/GoogleCloudPlatform/pubsub/pull/240/commits/07d5c3d0efbac1cd1d512ba006dc872933f6fe7d.

**Review by samarthsingal — APPROVED** [review]
LGTM!

**Pull request merged**